lib/compat/wordpress-6.3/theme-previews.php

```diff
@@ -112,13 +112,27 @@ function block_theme_activate_nonce() {
 	<?php
 }
 
+function gutenberg_render_block_block_theme_previews( $block_content, $parsed_block, $block_class ) {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

Would it be better to limit the execution of this function to only when previewing the front of the site (or not)? What do you think?

Reply:
We should check for permissions as we do in `gutenberg_get_theme_preview_path` so that the whole string manipulaiton only happens for logged is users previewing thiemes, otherwise we could just hit the site with bogus requests.